x-pack/plugins/alerting/server/lib/alert_execution_store.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { set } from 'lodash';
+import { ActionsConfigMap } from './get_actions_config_map';
+import { ActionsCompletion } from '../task_runner/types';
+
+interface State {
+  numberOfTriggeredActions: number;
+  numberOfGeneratedActions: number;
+  connectorTypes: {
+    [key: string]: {
+      triggeredActionsStatus: ActionsCompletion;
+      numberOfTriggeredActions: number;
+      numberOfGeneratedActions: number;
+    };
+  };
+}
+
+export class AlertExecutionStore {
+  private state: State = {
+    numberOfTriggeredActions: 0,
+    numberOfGeneratedActions: 0,
+    connectorTypes: {},
+  };
+
+  // Getters
+  public getTriggeredActionsStatus = () => {
+    const hasPartial = Object.values(this.state.connectorTypes).some(
+      (connectorType) => connectorType?.triggeredActionsStatus === ActionsCompletion.PARTIAL
+    );
+    return hasPartial ? ActionsCompletion.PARTIAL : ActionsCompletion.COMPLETE;
+  };
+  public getNumberOfTriggeredActions = () => {
+    return this.state.numberOfTriggeredActions;
+  };
+  public getNumberOfGeneratedActions = () => {
+    return this.state.numberOfGeneratedActions;
+  };
+  public getStatusByConnectorType = (actionTypeId: string) => {
+    return this.state.connectorTypes[actionTypeId];
+  };
+
+  // Setters
+  public setNumberOfTriggeredActions = (numberOfTriggeredActions: number) => {
+    this.state.numberOfTriggeredActions = numberOfTriggeredActions;
+  };
+
+  public setNumberOfGeneratedActions = (numberOfGeneratedActions: number) => {
+    this.state.numberOfGeneratedActions = numberOfGeneratedActions;
+  };
+
+  public setTriggeredActionsStatusByConnectorType = ({
+    actionTypeId,
+    status,
+  }: {
+    actionTypeId: string;
+    status: ActionsCompletion;
+  }) => {
+    set(this.state, `connectorTypes["${actionTypeId}"].triggeredActionsStatus`, status);
+  };
+
+  // Checkers
+  public hasReachedTheExecutableActionsLimit = (actionsConfigMap: ActionsConfigMap): boolean =>
```

Comment:
nit: seems like for this and the next `hasReached...()` method, we could pass the `actionsConfigMap()` into the constructor, so we wouldn't have to pass it into these methods directly.  Haven't seen where it's used, but it could save having to pass the configMap object around with this object ... OTOH, perhaps there is a reason we do this, or we already have it available and the current structure makes it easier to test ...

Reply:
i could be moved in to this class but they are completely different objects.
This class is to hold ruleRunMetrics actionsConfigMap is the config object from kibana.yml...
Maybe later we can do this refactoring, lets discuss :)